go/engine/puk_roll.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2017 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+// PerUserKeyRoll creates a new per-user-key for the active user.
+// This can be the first per-user-key for the user.
+package engine
+
+import (
+	"fmt"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+// PerUserKeyRoll is an engine.
+type PerUserKeyRoll struct {
+	libkb.Contextified
+	args      *PerUserKeyRollArgs
+	DidNewKey bool
+}
+
+type PerUserKeyRollArgs struct {
+	LoginContext libkb.LoginContext // optional
+	Me           *libkb.User        // optional
+}
+
+// NewPerUserKeyRoll creates a PerUserKeyRoll engine.
+func NewPerUserKeyRoll(g *libkb.GlobalContext, args *PerUserKeyRollArgs) *PerUserKeyRoll {
+	return &PerUserKeyRoll{
+		args:         args,
+		Contextified: libkb.NewContextified(g),
+	}
+}
+
+// Name is the unique engine name.
+func (e *PerUserKeyRoll) Name() string {
+	return "PerUserKeyRoll"
+}
+
+// GetPrereqs returns the engine prereqs.
+func (e *PerUserKeyRoll) Prereqs() Prereqs {
+	return Prereqs{
+		Session: true,
```

Comment:
Oh yeah, thanks, I'll get that as part of the rebase of the next one https://github.com/keybase/client/pull/7731